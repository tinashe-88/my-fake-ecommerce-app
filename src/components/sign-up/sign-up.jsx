import React from 'react'

import FormInput from '../form-input/form-input'
import CustomButton from '../custom-button/custom-button'

import {
  auth, createUserProfileDocument
} from '../../firebase/firebase.utils'

import {
  SignUpContainer,
  SignUpTitle
} from './sign-up.styles'

class SignUp extends React.Component {
  constructor(){
    super()

    this.state = {
      displayName: '',
      email: '',
      password: '',
      confirmPassword: ''
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
      this.setState({
        [name]: value
      })
  }

  handleSubmit = async (e) => {
    e.preventDefault()

    const {
      displayName,
      email,
      password,
      confirmPassword
    } = this.state

    if(password !== confirmPassword){
      alert('Passwords do not match. Please try again')
      return
    }

    try {
      const { user } = await auth.createUserWithEmailAndPassword(
        email, password
      )

      await createUserProfileDocument(user, { displayName })

      this.setState({
        displayName: '',
        email: '',
        password: '',
        confirmPassword: ''
      })
    } catch(err){
      console.error(err)
    }
  }

  render(){
    const { displayName, email, password, confirmPassword} = this.state
    return (
      <SignUpContainer>
        <SignUpTitle>Don't have a account?</SignUpTitle>
        <span>Sign up with your email and password</span>
        <form className='sign-up-form' onSubmit={this.handleSubmit}>
          <FormInput
            type='text'
            name='displayName'
            value={displayName}
            onChange={this.handleChange}
            label='Display Name'
            required
          />
          <FormInput
            type='email'
            name='email'
            value={email}
            onChange={this.handleChange}
            label='Email'
            required
          />
          <FormInput
            type='password'
            name='password'
            value={password}
            onChange={this.handleChange}
            label='Password'
            required
          />
          <FormInput
            type='password'
            name='confirmPassword'
            value={confirmPassword}
            onChange={this.handleChange}
            label='Confirm Password'
            required
          />
          <CustomButton type='submit'>SIGN UP</CustomButton>
        </form>
      </SignUpContainer>
    )
  }
}

export default SignUp