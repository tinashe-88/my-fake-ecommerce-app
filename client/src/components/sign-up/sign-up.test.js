import React from 'react'
import { shallow } from 'enzyme'

import { SignUp } from './sign-up'

it('Should render SignUp component', () => {
  expect(shallow(<SignUp />)).toMatchSnapshot()
})