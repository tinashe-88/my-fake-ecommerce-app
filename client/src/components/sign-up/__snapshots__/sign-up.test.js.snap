// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SignUp component 1`] = `
<styled.div>
  <styled.h2>
    Don't have a account?
  </styled.h2>
  <span>
    Sign up with your email and password
  </span>
  <form
    className="sign-up-form"
    onSubmit={[Function]}
  >
    <FormInput
      label="Display Name"
      name="displayName"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <FormInput
      label="Email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <FormInput
      label="Password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <FormInput
      label="Confirm Password"
      name="confirmPassword"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <CustomButton
      type="submit"
    >
      SIGN UP
    </CustomButton>
  </form>
</styled.div>
`;
